frontend/src/app/subject-comments.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/toPromise';
+import 'rxjs/add/operator/share';
+import { SubjectComment } from './subject-comment';
+
+@Injectable()
+export class SubjectCommentsService {
+    constructor(private http: Http) {}
+
+    fetch(university: number, subject: number): Promise<SubjectComment[]>{
+        return this.http
+            .get(`http://localhost:8080/api/universities/university/${university}/subjects/subject/${subject}/comments`)
+            .toPromise()
+            .then(response => {
+                return response.json() as SubjectComment[];
+            });
+    }
+
+    add(university: number, subject: number, message: string): void{
+        //unsure if this post works
+        this.http
```

Comment:
These should all be returning values and they should all be promises. The caller of these methods won't be able to see what the responses are. Check out my pull request and see how I've handled my POST requests.

Reply:
"should all" means, including thumbs and flags methods?